cityofaustin/atd-prefect
# Upgrade service bot intake issue flow to Prefect 2

https://github.com/cityofaustin/atd-data-tech/issues/11418

This PR migrates the service bot intake (Knack service request to Github issue) flow to Prefect 2. Going to branch from here for the other two flows to keep this from getting too big and hard to test.

I tried out Secret blocks and also attempted staging and production runs/environments. Prod is default but you can pass `env` (new parameter) and `docker_tag` (existing parameter) to use the staging Secret block and test an updated image if needed. Super open to feedback on that since not all flows will have a clear staging env like this one since it has a staging/test Knack app.

Concurrency limit is set to 1 as [mentioned here](https://austininnovation.slack.com/archives/CHZE6BC6L/p1678745191242269) - still need to discuss how to clear the queue of late flows in the next Prefect meeting

### Testing

1. Submit a service request in the Knack app called `DTS | Data and Technology Services Portal | PRODUCTION Copy` which I've linked below but before that:
         - Be sure to name the request something obvious like "TEST Service Bot issue" because it will create a real issue [here](https://github.com/cityofaustin/atd-data-tech/issues)
         - Be sure to check that the url matches the one below and that you are not creating a new service request in the production app or else it will get picked up by the production bot instead of this Prefect 2 one (sometimes clicking links in Knack apps can redirect you to the production app without you realizing it)
         - https://atd.knack.com/dts--data-and-technology-services-portal--production-copy#new-service-request/ 
         - Also be aware that this will end up triggering an email that goes to the transportation.data inbox which could be confusing to the service desk primary. I've given Andrew S. a heads up about testing this week.

3. Go to the Prefect 2 cloud https://app.prefect.cloud/ and sign in with transportation.data@austintexas.gov which sends a magic link. After you are signed in go to https://app.prefect.cloud/account/62be21ad-87d4-4858-bc47-a49e1fd29df7/workspace/4e909dab-3495-4e26-9403-9b633af2c970/deployments/deployment/4e3e7729-43bf-46da-a0ee-8fed2d11e9cd

4. In the top right, click the **Run** button and choose **Custom run** from the dropdown
5. Double check the Parameters at the bottom of this view show `staging` as the value for `env` and then click **Run**
6. You can watch the logs to make sure the flow succeeds and then go to https://github.com/cityofaustin/atd-data-tech/issues
7. You should see your service request in the form of a new Github issue which you can close
8. Clean out the emails that this generated in transportation.data inbox
